lib/saluki-components/src/sources/checks/builder/python/python_modules.rs

```diff
@@ -65,20 +60,37 @@ fn try_send_event(event: EventD) -> Result<(), GenericError> {
 }
 
 fn get_config_key(key: String) -> String {
-    match GLOBAL_CONFIGURATION.get() {
-        Some(configuration) => configuration.get_typed_or_default::<String>(&key),
-
+    match GLOBAL_EXECUTION_CONTEXT.get() {
+        Some(execution_context) => execution_context.configuration().get_typed_or_default::<String>(&key),
         None => "".to_string(),
     }
 }
 
 fn fetch_hostname() -> &'static str {
-    match GLOBAL_HOST.get() {
-        Some(host) => host.as_str(),
+    match GLOBAL_EXECUTION_CONTEXT.get() {
+        Some(execution_context) => execution_context.hostname(),
         None => "",
     }
 }
 
+fn fetch_tracemalloc_enabled() -> bool {
+    match GLOBAL_EXECUTION_CONTEXT.get() {
+        Some(execution_context) => execution_context
+            .configuration()
+            .get_typed_or_default::<bool>("tracemalloc_debug"),
+        None => false,
+    }
+}
```

Comment:
Maybe we can use a generic `get_config_key` and use it to fetch `tracemalloc_debug` instead of having a dedicated function for `fetch_tracemalloc_enabled`

Something like: 

```rust 
fn get_config_key<'a, T>(key: &str) -> T
where
    T: Default + Deserialize<'a>,
{
    match GLOBAL_EXECUTION_CONTEXT.get() {
        Some(execution_context) => execution_context.configuration().get_typed_or_default::<T>(key),
        None => T::default(),
    }
}
```

Note, that I'm not sure if this idiomatic Rust, better as @tobz 🙏

Reply:
That sound like a very good idea! Implemented in the last commit.